nipype/pipeline/engine/tests/test_engine.py

```diff
@@ -518,35 +509,26 @@ def func2(a):
     w1.config['execution'] = {'stop_on_first_crash': 'true',
                               'local_hash_check': 'false',
                               'crashdump_dir': wd}
-    error_raised = False
     # create dummy distributed plugin class
     from nipype.pipeline.plugins.base import DistributedPluginBase
 
     class RaiseError(DistributedPluginBase):
         def _submit_job(self, node, updatehash=False):
             raise Exception('Submit called')
-    try:
+
+    with pytest.raises(Exception) as excinfo:
```

Comment:
I'm not sure if I understand. Do you want pytest to check specific exceptions or you want to define a class `EngineTestException(Exception)`.

I sometimes ask pytest to check a specific error or exception, but in this case the exception from line 517 was raised (i.e. `raise Exception('Submit called')`), so to be more specific I could only check the message and I've done it in the line 521 `assert 'Submit called' == str(excinfo.value)`.